.github/workflows/release.yml

```diff
@@ -5,6 +5,10 @@ on:
     branches:
       - master
 
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
```

Comment:
I think the release workflow, specifically, should probably work the same way that the meson repo's documentation workflow `website.yml` works:

```yml
concurrency:
  group: mesonbuild.com
  # We do NOT want `cancel-in-progress` here since only one website job
  # should run at a time to avoid upload race conditions.
```